Automattic/pocket-casts-ios
# Bookmarks: Add the bookmark list to the player tab

This adds the base bookmarks list design to the player tab and adds pulling of real data from the database,

There's still a few items to add which will be done in a future PR:
- Multi select
- Empty state
- Navigational routing
- Editing/Deleting

## Screenshots

| iPod Touch | iPhone 14 Pro | iPad |
|:---:|:---:|:---:|
|<img width="160" src="https://github.com/Automattic/pocket-casts-ios/assets/793774/bfde5445-304e-47bb-a19a-54dc70705ca2">|<img width="160" src="https://github.com/Automattic/pocket-casts-ios/assets/793774/69e310d7-35b8-4bc6-b3c8-add4da836d69">|<img width="160" src="https://github.com/Automattic/pocket-casts-ios/assets/793774/3305bba2-4518-4d30-8b39-4746b45c3915">|


## To test
1. Enable the Bookmarks Feature Flag by going to Profile > Cog > Beta Features > bookmarks
2. Open the Full Screen Player
3. Swipe over to Bookmarks
4. ✅ Verify you see an empty list that says 0 bookmarks
5. Swipe over to the Now Playing
6. Tap the ... button in the player shelf
7. Tap Edit
8. Move the Add Bookmark item to the top of the list
9. Tap done
10. Tap the Add Bookmark item
11. Swipe to the Bookmarks tab
12. ✅ Verify you see the bookmark and the title is the correct timestamp
13. Go back to Now Playing, move to a new position in the episode. Tap the Add Bookmark action
14. Go back to Bookmarks, ✅ Verify it appears
15. Dismiss the player
16. Reopen the player
17. Swipe to the Bookmarks tab
18. ✅ Verify it appears
19. Pause the playing episode
20. Tap the Play button on the bookmarks list
21. ✅ Verify it plays at the correct time. 
22. Switch themes, and ✅ Verify it looks good on each theme.


## Checklist
